addons/sale_project/models/project.py

```diff
@@ -823,22 +821,26 @@ def _group_expand_sales_order(self, sales_orders, domain, order):
             return search_on_comodel
         return sales_orders
 
-    @api.depends('sale_line_id', 'project_id', 'partner_id.commercial_partner_id', 'allow_billable')
+    @api.depends('sale_line_id', 'project_id.sale_order_id', 'partner_id.commercial_partner_id', 'allow_billable')
     def _compute_sale_order_id(self):
         for task in self:
             if not task.allow_billable:
                 task.sale_order_id = False
                 continue
-            sale_order_id = task.sale_order_id or self.env["sale.order"]
-            if task.sale_line_id:
-                sale_order_id = task.sale_line_id.sudo().order_id
-            elif task.project_id.sale_order_id:
-                sale_order_id = task.project_id.sale_order_id
-            if task.partner_id.commercial_partner_id != sale_order_id.partner_id.commercial_partner_id:
-                sale_order_id = False
-            if sale_order_id and not task.partner_id:
-                task.partner_id = sale_order_id.partner_id
```

Comment:
the change also remove the setting of the task partner_id, is this intended too?

Reply:
Yes, because that code was unreachable anyway:

- `sale_order_id` has to exist
- `task.partner_id` has to be false
- if `sale_order_id` exists, then so does `sale_order_id.partner_id.commercial_partner_id`
- if `task.partner_id` is false, then so is `task.partner_id.commercial_partner_id`


Consequently, with `sale_order_id` existing, and `task.partner_id` being false, the preceding conditional will always set `sale_order_id` to false as well, because their commercial partner fields always differ.

Hence it is impossible for `sale_order_id and not task.partner_id` to ever succeed at that point in order to set the `task.partner_id` field, because if that field were to be false, then `sale_order_id` will always be false as well.